modules/openapi-generator/src/main/resources/dart2/api_client.mustache

```diff
@@ -1,182 +1,187 @@
-                             String nullableContentType,
-                             List<String> authNames) async {
+  T getAuthentication<T extends Authentication>(String name) {
+    final authentication = _authentications[name];
+    return authentication is T ? authentication : null;
+  }
 
+  // We don’t use a Map<String, String> for queryParams.
+  // If collectionFormat is "multi" a key might appear multiple times.
+  Future<Response> invokeAPI(
+    String path,
+    String method,
+    Iterable<QueryParam> queryParams,
+    Object body,
+    Map<String, String> headerParams,
+    Map<String, String> formParams,
+    String nullableContentType,
+    List<String> authNames,
+  ) async {
     _updateParamsForAuth(authNames, queryParams, headerParams);
 
-    var ps = queryParams
+    headerParams.addAll(_defaultHeaderMap);
+
+    final ps = queryParams
       .where((p) => p.value != null)
-      .map((p) => '${p.name}=${Uri.encodeQueryComponent(p.value)}');
+      .map((p) => "${p.name}=${Uri.encodeQueryComponent(p.value)}");
 
-    String queryString = ps.isNotEmpty ?
-                         '?' + ps.join('&') :
-                         '';
+    final queryString = ps.isNotEmpty ? "?" + ps.join("&") : "";
 
-    String url = basePath + path + queryString;
+    final url = "$basePath$path$queryString";
 
-    headerParams.addAll(_defaultHeaderMap);
     if (nullableContentType != null) {
-      final contentType = nullableContentType;
-      headerParams['Content-Type'] = contentType;
+      headerParams["Content-Type"] = nullableContentType;
     }
 
-    if(body is MultipartRequest) {
+    if (body is MultipartRequest) {
       var request = MultipartRequest(method, Uri.parse(url));
       request.fields.addAll(body.fields);
       request.files.addAll(body.files);
       request.headers.addAll(body.headers);
       request.headers.addAll(headerParams);
       var response = await client.send(request);
       return Response.fromStream(response);
-    } else {
-      var msgBody = nullableContentType == "application/x-www-form-urlencoded" ? formParams : serialize(body);
-      final nullableHeaderParams = (headerParams.isEmpty)? null: headerParams;
-      switch(method) {
-        case "POST":
-          return client.post(url, headers: nullableHeaderParams, body: msgBody);
-        case "PUT":
-          return client.put(url, headers: nullableHeaderParams, body: msgBody);
-        case "DELETE":
-          return client.delete(url, headers: nullableHeaderParams);
-        case "PATCH":
-          return client.patch(url, headers: nullableHeaderParams, body: msgBody);
-        case "HEAD":
-          return client.head(url, headers: nullableHeaderParams);
+    }
+
+    final msgBody = nullableContentType == "application/x-www-form-urlencoded"
+      ? formParams
+      : serialize(body);
+    final nullableHeaderParams = headerParams.isEmpty ? null : headerParams;
+
+    try {
+      switch(method.toUpperCase()) {
```

Comment:
I'm not convinced we should be correcting method type. Most languages do not do it, found it in only 2 other templates.

Reply:
Agreed. Removing it.